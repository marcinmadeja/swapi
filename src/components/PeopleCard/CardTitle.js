import React from 'react';
import PersonIcon from 'material-ui-icons/Person';

import {
  Title,
  TitlePicture,
  UserName,
} from './CardTitle.styles';

const CardTitle = ({ name }) => {
  return (
    <Title>
      <TitlePicture>
        <PersonIcon />
      </TitlePicture>

      <UserName>{name}</UserName>
    </Title>
  );
};

export default CardTitle;
