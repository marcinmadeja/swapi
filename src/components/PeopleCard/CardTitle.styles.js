import styled from 'styled-components';

export const Title = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`;

export const TitlePicture = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 60px;
  height: 60px;
  margin: 0 auto;
  border-radius: 100%;

  background: rgba(0,0,0,0.04);
  color: rgba(0,0,0,0.3);

  svg {
    width: 30px;
    height: 30px;
  }
`;

export const UserName = styled.h4`
  margin: 15px 0 0;
  font-size: 18px;
  font-weight: 500;
`;
