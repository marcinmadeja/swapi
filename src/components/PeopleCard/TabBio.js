import React from 'react';

import {
  Content,
  Item,
  Label,
  Value,
} from './TabBio.styles';

const TabBio = ({
  userData: {
    birth_year,
    eye_color,
    hair_color,
    height,
    mass,
    skin_color,
  },
}) => {
  return (
    <Content>
      <Item>
        <Label>Birthday:</Label>
        <Value>{birth_year}</Value>
      </Item>

      <Item>
        <Label>Eye color:</Label>
        <Value>{eye_color}</Value>
      </Item>

      <Item>
        <Label>hair color:</Label>
        <Value>{hair_color}</Value>
      </Item>

      <Item>
        <Label>Height:</Label>
        <Value>{height}</Value>
      </Item>

      <Item>
        <Label>Mass:</Label>
        <Value>{mass}</Value>
      </Item>

      <Item>
        <Label>Skin color:</Label>
        <Value>{skin_color}</Value>
      </Item>
    </Content>
  );
};

export default TabBio;
