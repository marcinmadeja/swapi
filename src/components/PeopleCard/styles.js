import styled from 'styled-components';

export const List = styled.ul`

`;

export const ListItem = styled.li`
  line-height: 1.8;
  letter-spacing: 0.7px;
`;
