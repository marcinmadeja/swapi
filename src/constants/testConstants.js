export const users = [
  {
    gender: 'male',
    name: {
      title: 'mr',
      first: 'terrence',
      last: 'watts',
    },
    location: {
      street: '5796 rolling green rd',
      city: 'bowral',
      state: 'victoria',
      postcode: 7762,
      coordinates: {
        latitude: '-79.9438',
        longitude: '-160.0279',
      },
      timezone: {
        offset: '-7:00',
        description: 'Mountain Time (US & Canada)',
      },
    },
    email: 'terrence.watts@example.com',
    login: {
      uuid: 'ef30b2c2-7575-42d1-b9e5-38030d64b184',
      username: 'tinygoose275',
      password: 'makaveli',
      salt: 'EwEwiYhD',
      md5: '67600680d1f62c990021d597e96b264d',
      sha1: '174b563d3f42d98f88e7bcaad064e647ee726faa',
      sha256: 'fc93dd24693ad0650413b01980ccc494dd3921991a8ab41e45cadf8610aa046a',
    },
    dob: {
      date: '1972-09-08T12:37:28Z',
      age: 45,
    },
    registered: {
      date: '2003-03-25T20:55:50Z',
      age: 15,
    },
    phone: '00-8437-1526',
    cell: '0476-938-903',
    id: {
      name: 'TFN',
      value: '686111438',
    },
    picture: {
      large: 'https://randomuser.me/api/portraits/men/24.jpg',
      medium: 'https://randomuser.me/api/portraits/med/men/24.jpg',
      thumbnail: 'https://randomuser.me/api/portraits/thumb/men/24.jpg',
    },
    nat: 'AU',
  },
  {
    gender: 'male',
    name: {
      title: 'mr',
      first: 'paulo',
      last: 'seyfarth',
    },
    location: {
      street: 'talstra√üe 94',
      city: 'bogen',
      state: 'bayern',
      postcode: 74635,
      coordinates: {
        latitude: '18.2360',
        longitude: '72.7154',
      },
      timezone: {
        offset: '-3:00',
        description: 'Brazil, Buenos Aires, Georgetown',
      },
    },
    email: 'paulo.seyfarth@example.com',
    login: {
      uuid: 'af5ee15a-da22-4264-8c94-7cd57e21bf97',
      username: 'crazybutterfly740',
      password: 'colt45',
      salt: '7LDAWnJY',
      md5: 'e5bbaee291478ce51472ec82d31f97c6',
      sha1: 'ea8722d0fc4ba69dc43e36661468f5bda5199799',
      sha256: 'e31a953387293ab446bf702c20013a91f1c91793fe59fe425a6e6502d502ad50',
    },
    dob: {
      date: '1969-06-18T22:13:53Z',
      age: 49,
    },
    registered: {
      date: '2004-09-15T22:48:16Z',
      age: 13,
    },
    phone: '0477-2629824',
    cell: '0175-1061186',
    id: {
      name: '',
      value: null,
    },
    picture: {
      large: 'https://randomuser.me/api/portraits/men/98.jpg',
      medium: 'https://randomuser.me/api/portraits/med/men/98.jpg',
      thumbnail: 'https://randomuser.me/api/portraits/thumb/men/98.jpg',
    },
    nat: 'DE',
  },
  {
    gender: 'male',
    name: {
      title: 'mr',
      first: 'mathias',
      last: 'mortensen',
    },
    location: {
      street: '9165 fredensvej',
      city: 'pandrup',
      state: 'nordjylland',
      postcode: 39076,
      coordinates: {
        latitude: '-12.9436',
        longitude: '101.5652',
      },
      timezone: {
        offset: '+7:00',
        description: 'Bangkok, Hanoi, Jakarta',
      },
    },
    email: 'mathias.mortensen@example.com',
    login: {
      uuid: '39d341b7-2f7b-4b4a-bfd4-6921922da746',
      username: 'beautifulladybug715',
      password: 'carpente',
      salt: 'KpPfkc93',
      md5: '5f976528c474e2c1479464183d7800d9',
      sha1: '6d25ac15f23bf114e05cb940e7c61925b3274746',
      sha256: 'cf715b8dc50773efd203d755f2b0c268fd588e5b98204545ad1a71d30b66e4b2',
    },
    dob: {
      date: '1990-04-03T15:42:14Z',
      age: 28,
    },
    registered: {
      date: '2014-05-19T01:46:06Z',
      age: 4,
    },
    phone: '04223613',
    cell: '62185716',
    id: {
      name: 'CPR',
      value: '722632-7968',
    },
    picture: {
      large: 'https://randomuser.me/api/portraits/men/13.jpg',
      medium: 'https://randomuser.me/api/portraits/med/men/13.jpg',
      thumbnail: 'https://randomuser.me/api/portraits/thumb/men/13.jpg',
    },
    nat: 'DK',
  },
];

export const swapiPeople = [
  {
    name: 'Leia Organa',
    height: '150',
    mass: '49',
    hair_color: 'brown',
    skin_color: 'light',
    eye_color: 'brown',
    birth_year: '19BBY',
    gender: 'female',
    homeworld: 'https://swapi.co/api/planets/2/',
    films: [
      'https://swapi.co/api/films/2/',
      'https://swapi.co/api/films/6/',
      'https://swapi.co/api/films/3/',
      'https://swapi.co/api/films/1/',
      'https://swapi.co/api/films/7/',
    ],
    species: [
      'https://swapi.co/api/species/1/',
    ],
    vehicles: [
      'https://swapi.co/api/vehicles/30/',
    ],
    starships: [],
    created: '2014-12-10T15:20:09.791000Z',
    edited: '2014-12-20T21:17:50.315000Z',
    url: 'https://swapi.co/api/people/5/',
  },
  {
    name: 'Obi-Wan Kenobi',
    height: '182',
    mass: '77',
    hair_color: 'auburn, white',
    skin_color: 'fair',
    eye_color: 'blue-gray',
    birth_year: '57BBY',
    gender: 'male',
    homeworld: 'https://swapi.co/api/planets/20/',
    films: [
      'https://swapi.co/api/films/2/',
      'https://swapi.co/api/films/5/',
      'https://swapi.co/api/films/4/',
      'https://swapi.co/api/films/6/',
      'https://swapi.co/api/films/3/',
      'https://swapi.co/api/films/1/',
    ],
    species: [
      'https://swapi.co/api/species/1/',
    ],
    vehicles: [
      'https://swapi.co/api/vehicles/38/',
    ],
    starships: [
      'https://swapi.co/api/starships/48/',
      'https://swapi.co/api/starships/59/',
      'https://swapi.co/api/starships/64/',
      'https://swapi.co/api/starships/65/',
      'https://swapi.co/api/starships/74/',
    ],
    created: '2014-12-10T16:16:29.192000Z',
    edited: '2014-12-20T21:17:50.325000Z',
    url: 'https://swapi.co/api/people/10/',
  },
];

