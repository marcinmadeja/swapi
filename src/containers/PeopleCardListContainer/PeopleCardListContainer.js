import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { requestSwPeople } from 'actions/swPeople';
import { updateSwFilms, updateSwVehicles, updateSwStarships } from 'actions/swUpdate';

import PeopleCardList from 'components/PeopleCardList/PeopleCardList';
import PeopleCard from 'components/PeopleCard/PeopleCard';
import { AlertWarning, AlertStandard } from 'components/alerts';

export class PeopleCardListContainer extends Component {
  componentDidMount() {
    this.props.requestSwPeople();
  }

  render() {
    const {
      list,
      pending,
      errors,

      filmsList,
      starshipsList,
      vehiclesList,
    } = this.props;

    if (pending) return <AlertStandard msg="Loading data" progressBar />;
    if (errors) return <AlertWarning msg="There was an error" />;
    if (!list.length) return null;
    return (
      <PeopleCardList>
        {list.map(user => (
          <PeopleCard
            key={user.url}

            user={user}
            filmsList={filmsList}
            starshipsList={starshipsList}
            vehiclesList={vehiclesList}

            updateSwFilms={this.props.updateSwFilms}
            updateSwVehicles={this.props.updateSwVehicles}
            updateSwStarships={this.props.updateSwStarships}
          />
        ))}
      </PeopleCardList>
    );
  }
}

const mapStateToProps = state => {
  const { list, pending, errors } = state.swPeople;
  const filmsList = state.swFilms.list;
  const starshipsList = state.swStarships.list;
  const vehiclesList = state.swVehicles.list;

  return {
    list,
    pending,
    errors,
    filmsList,
    starshipsList,
    vehiclesList,
  };
};

PeopleCardListContainer.propTypes = {
  list: PropTypes.array.isRequired,
  pending: PropTypes.bool.isRequired,
  errors: PropTypes.bool.isRequired,

  filmsList: PropTypes.array.isRequired,
  starshipsList: PropTypes.array.isRequired,
  vehiclesList: PropTypes.array.isRequired,

  requestSwPeople: PropTypes.func.isRequired,
  updateSwFilms: PropTypes.func.isRequired,
  updateSwVehicles: PropTypes.func.isRequired,
  updateSwStarships: PropTypes.func.isRequired,
};

export default connect(
  mapStateToProps,
  { requestSwPeople, updateSwFilms, updateSwVehicles, updateSwStarships },
)(PeopleCardListContainer);
