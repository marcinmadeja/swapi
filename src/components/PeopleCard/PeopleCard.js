import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Col } from 'components/grid';

import { withStyles } from 'material-ui/styles';
import Card from 'material-ui/Card';

import TabsHeader from './TabsHeader';
import CardTitle from './CardTitle';
import TabsContent from './TabsContent';


const styles = (theme) => ({
  card: {
    marginBottom: 30,
    padding: 10,
  },
  tabRoot: {
    minWidth: 'auto',
    flex: 1,
    height: 60,
  },
  tabsRoot: {
    background: 'rgba(0,0,0,0.01)',
    margin: '10px -10px 15px',
    boxShadow: theme.shadows[2],
  },
});

class PeopleCard extends Component {
  constructor(props) {
    super(props);

    this.state = { currentTab: 0 };
    this.handleTabChange = this.handleTabChange.bind(this);
  }

  handleTabChange(event, value) {
    this.setState({ currentTab: value });
  }

  render() {
    const {
      classes,
      user,
    } = this.props;

    const { currentTab } = this.state;

    return (
      <Col md={4}>
        <Card className={classes.card}>
          <CardTitle name={user.name} />

          <TabsHeader
            currentTab={currentTab}
            onTabChange={this.handleTabChange}
            classes={classes}
          />

          <TabsContent
            currentTab={currentTab}
            {...this.props}
          />
        </Card>
      </Col>
    );
  }
}

PeopleCard.propTypes = {
  user: PropTypes.object.isRequired,
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(PeopleCard);
