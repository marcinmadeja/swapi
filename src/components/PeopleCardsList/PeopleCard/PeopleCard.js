import React, { Component } from 'react';
import { connect } from 'react-redux';
import { updateSwFilms } from 'actions/swUpdate';
import { compose } from 'redux';
import { Col } from 'components/grid';

import SwipeableViews from 'react-swipeable-views';
import { withStyles } from 'material-ui/styles';
import Card from 'material-ui/Card';
import Tabs, { Tab } from 'material-ui/Tabs';
import PhoneIcon from 'material-ui-icons/Phone';

import PersonIcon from 'material-ui-icons/Person';
import DirectionsCarIcon from 'material-ui-icons/DirectionsCar';
import LocalAirportIcon from 'material-ui-icons/LocalAirport';
import LocalMoviesIcon from 'material-ui-icons/LocalMovies';

import TabContent from './TabContent';
import TabBio from './TabBio';
import TabFilms from './TabFilms';
import TabStarships from './TabStarships';
import TabVehicles from './TabVehicles';

import {
  styles,
  Title,
  TitlePicture,
  UserName,
} from './PeopleCard.styles';

class PeopleCard extends Component {
  constructor(props) {
    super(props);

    this.state = {
      currentTab: 0,
    };

    this.handleTabChange = this.handleTabChange.bind(this);
  }

  handleTabChange(event, value) {
    this.setState({ currentTab: value });
  }

  render() {
    const {
      classes,
      user,
    } = this.props;

    const { currentTab } = this.state;

    return (
      <Col md={4}>
        <Card className={classes.card}>
          <Title>
            <TitlePicture>
              <PersonIcon />
            </TitlePicture>

            <UserName>{user.name}</UserName>
          </Title>

          <Tabs
            value={currentTab}
            onChange={this.handleTabChange}
            indicatorColor="primary"
            textColor="primary"
            className={classes.tabsRoot}
          >
            <Tab className={classes.tabRoot} label="Bio" icon={<PhoneIcon />} />
            <Tab className={classes.tabRoot} label="Films" icon={<LocalMoviesIcon />} />
            <Tab className={classes.tabRoot} label="Starships" icon={<LocalAirportIcon />} />
            <Tab className={classes.tabRoot} label="Vehicles" icon={<DirectionsCarIcon />} />
          </Tabs>


          <SwipeableViews
            index={currentTab}
            onChangeIndex={this.handleChangeIndex}
          >
            <TabContent forceLoaded render={() => (<TabBio userData={user} />)} />

            <TabContent
              name={user.name}
              isActive={currentTab === 1}
              urlList={user.films}
              loadedData={this.props.filmsList}
              updateData={this.props.updateSwFilms}
              render={() => (<TabFilms userData={user} />)}
            />

            <TabContent render={() => (<TabStarships userData={user} />)} />

            <TabContent render={() => (<TabVehicles userData={user} />)} />
          </SwipeableViews>
        </Card>
      </Col>
    );
  }
}

const mapStateToProps = state => {
  const filmsList = state.swFilms.list;
  return {
    filmsList,
  };
};

export default compose(
  connect(mapStateToProps, { updateSwFilms }),
  withStyles(styles),
)(PeopleCard);
