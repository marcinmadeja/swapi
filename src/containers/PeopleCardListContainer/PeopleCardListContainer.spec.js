import React from 'react';
import { swapiPeople } from 'constants/testConstants';
import PeopleCard from 'components/PeopleCard/PeopleCard';
import { PeopleCardListContainer } from './PeopleCardListContainer';

const defaultProps = {
  list: [],
  pending: false,
  errors: false,

  filmsList: [],
  starshipsList: [],
  vehiclesList: [],
};

const setup = (props = {}) => {
  props = { ...defaultProps, ...props };
  const actions = {
    requestSwPeople: jest.fn(),
    updateSwFilms: jest.fn(),
    updateSwVehicles: jest.fn(),
    updateSwStarships: jest.fn(),
  };

  const component = <PeopleCardListContainer key={1} {...actions} {...props} />;
  const shallowComponent = shallow(component);

  return {
    actions,
    component,
    shallowComponent,
  };
};

describe('PeopleCardListContainer', () => {
  it('should render', () => {
    const { shallowComponent } = setup();
    expect(shallowComponent.exists()).toBe(true);
  });

  it('should show Loading data alert when pending', () => {
    const { component } = setup({ pending: true });

    const tree = renderer
      .create(component)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should show errors alert when something goes wrong', () => {
    const { component } = setup({ errors: true });

    const tree = renderer
      .create(component)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should return null when list array is empty', () => {
    const { shallowComponent } = setup({ list: [] });
    expect(shallowComponent.find(PeopleCard).length).toEqual(0);
  });

  it('should render PeopleCards list when non empty array was passed', () => {
    const { shallowComponent } = setup({ list: swapiPeople });
    expect(shallowComponent.find(PeopleCard).length).toEqual(swapiPeople.length);
  });
});

