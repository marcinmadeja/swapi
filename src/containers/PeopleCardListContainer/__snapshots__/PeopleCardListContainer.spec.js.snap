// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleCardListContainer should show Loading data alert when pending 1`] = `
<div
  className="sc-gzVnrw khfqoz"
>
  <div
    className="sc-gZMcBi jOpOnU"
  >
    <div
      className="MuiCircularProgress-root-1 MuiCircularProgress-colorPrimary-2"
      role="progressbar"
      style={
        Object {
          "height": 25,
          "width": 25,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg-4 MuiCircularProgress-svgIndeterminate-5"
        viewBox="0 0 50 50"
      >
        <circle
          className="MuiCircularProgress-circle-6 MuiCircularProgress-circleIndeterminate-7"
          cx={25}
          cy={25}
          fill="none"
          r={20}
          strokeWidth={2}
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  Loading data
</div>
`;

exports[`PeopleCardListContainer should show errors alert when something goes wrong 1`] = `
<div
  className="sc-dnqmqq chIbYT"
>
  There was an error
</div>
`;
